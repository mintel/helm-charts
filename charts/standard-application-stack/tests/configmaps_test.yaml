suite: Test ConfigMaps
templates:
  - configmaps.yaml
release:
  namespace: test-namespace
tests:
  - it: Empty configmap
    set:
      global.name: test-app
      configMaps:
        - name: "config"
          create: true
          configs: []
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: ConfigMap
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: test-app-config

  - it: Single singleline data entry
    set:
      global.name: test-app
      configMaps:
        - name: "config"
          create: true
          configs:
            - name: test1
              data: test-data
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - equal:
          path: data["test1"]
          value: test-data

  - it: Single multiline data entry
    set:
      global.name: test-app
      configMaps:
        - name: "config"
          create: true
          configs:
            - name: test1
              data: |
                test data line 1
                test data line 2
                test data line 3
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - equal:
          path: data["test1"]
          value: |
            test data line 1
            test data line 2
            test data line 3

  - it: Mixed singleline and  multiline data entries
    set:
      global.name: test-app
      configMaps:
        - name: "config"
          create: true
          configs:
            - name: test1
              data: |
                test data line 1
                test data line 2
                test data line 3
            - name: test2
              data: even more test data
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - equal:
          path: data["test1"]
          value: |
            test data line 1
            test data line 2
            test data line 3
      - equal:
          path: data["test2"]
          value: even more test data
