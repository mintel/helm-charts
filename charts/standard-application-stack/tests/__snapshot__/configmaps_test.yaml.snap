Empty configmap:
  1: |
    apiVersion: v1
    data: null
    kind: ConfigMap
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: config
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app-config
        app.mintel.com/application: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app-config
      name: test-app-config
      namespace: test-namespace
Mixed singleline and  multiline data entries:
  1: |
    apiVersion: v1
    data:
      test1: |
        test data line 1
        test data line 2
        test data line 3
      test2: even more test data
    kind: ConfigMap
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: config
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app-config
        app.mintel.com/application: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app-config
      name: test-app-config
      namespace: test-namespace
Single multiline data entry:
  1: |
    apiVersion: v1
    data:
      test1: |
        test data line 1
        test data line 2
        test data line 3
    kind: ConfigMap
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: config
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app-config
        app.mintel.com/application: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app-config
      name: test-app-config
      namespace: test-namespace
Single singleline data entry:
  1: |
    apiVersion: v1
    data:
      test1: test-data
    kind: ConfigMap
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: config
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app-config
        app.mintel.com/application: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app-config
      name: test-app-config
      namespace: test-namespace
ensures ArgoCD annotations are populated:
  1: |
    apiVersion: v1
    data: null
    kind: ConfigMap
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/hook: PreSync
        argocd.argoproj.io/hook-delete-policy: HookSucceeded
        argocd.argoproj.io/sync-wave: "-1"
      labels:
        app.kubernetes.io/component: config
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app-config
        app.mintel.com/application: example-app
        app.mintel.com/env: test
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app-config
      name: example-app-config
      namespace: test-namespace
